modules/seqexec/server/src/main/scala/seqexec/server/ghost/GhostController.scala

```diff
@@ -11,53 +11,12 @@ import giapi.client.commands.Configuration
 import giapi.client.ghost.GhostClient
 import giapi.client.GiapiConfig
 import giapi.client.syntax.giapiconfig._
-import seqexec.server.keywords.GdsClient
 import scala.concurrent.duration._
-import seqexec.server.ConfigUtilOps.{ContentError, ExtractFailure}
-import seqexec.server.GiapiInstrumentController
-import seqexec.server.ghost.GhostController.GhostConfig
-
-trait GhostController[F[_]] extends GiapiInstrumentController[F, GhostConfig, GhostClient[F]]
-
-object GhostController {
-  def apply[F[_]: Sync](clien: GhostClient[F], gdsClien: GdsClient[F]): GhostController[F] =
-    new GhostController[F] {
-      override val client = clien
-
-      override val gdsClient = gdsClien
-
-      override val name = "GHOST"
-
-      override def configuration(config: GhostConfig): F[Configuration] = config.configuration.pure[F]
-    }
-
-  private def ifuConfig(ifuNum: IFUNum,
-                        ifuTargetType: IFUTargetType,
-                        coordinates: Coordinates,
-                        bundleConfig: BundleConfig): Configuration = {
-    def cfg[P: GiapiConfig](paramName: String, paramVal: P) =
-      Configuration.single(s"${ifuNum.configValue}.$paramName", paramVal)
-      cfg("target", ifuTargetType.configValue) |+|
-      cfg("type", DemandType.DemandRADec.demandType) |+|
-      cfg("ra", coordinates.ra.toAngle.toDoubleDegrees) |+|
-      cfg("dec", coordinates.dec.toAngle.toDoubleDegrees) |+|
-      cfg("bundle", bundleConfig.configValue)
-  }
-
-  private def ifuPark(ifuNum: IFUNum): Configuration = {
-    def cfg[P: GiapiConfig](paramName: String, paramVal: P) =
-      Configuration.single(s"${ifuNum.ifuStr}.$paramName", paramVal)
-      cfg("target", IFUTargetType.NoTarget.targetType) |+|
-      cfg("type", DemandType.DemandPark.demandType)
-  }
-
-  sealed abstract class BundleConfig(val configName: String) {
-    def determineType(t: IFUTargetType): BundleConfig = t match {
-      case IFUTargetType.SkyPosition => BundleConfig.Sky
-      case _                         => this
-    }
-  }
+import seqexec.server.ConfigUtilOps.ContentError
+import seqexec.server.ConfigUtilOps.ExtractFailure
+import seqexec.server.keywords.GdsClient
 
+package ghost {
```

Comment:
It's weird to put this stuff in package blocks and in the package object … can we pull out classes into their own source files and put `Eq[GhostConfig]` into the companion for `GhostConfig`?

Reply:
Sure, I think package objects are going to be removed in dotty right?

Reply:
Yeah, but also `package foo { ... }` is virtually unknown and unused even though the language allows it.